### PR TITLE
Fix dark welcome

### LINKED ISSUE #892: Welcome screen does not render correctly in dark mode
See the screenshot below, the colors are not correct. I use a bigger font size, and the screen is scrollable so this can be ignored.

image

Screenshot ref. seems correct though: https://github.com/vector-im/element-x-android/blob/develop/tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests%5Bio.element.android.features.ftue.impl.welcome_null_DefaultGroup_WelcomeViewPreview-N-0_2_null,NEXUS_5,1.0,en%5D.png

### PR DESCRIPTION
**Pull request by bmarty**
Not sure why, but at runtime the color of the Text in Welcome Screen (and analtytics opt-in screen) were not correct (see screenshot in #892 )

The fixes is not visible in the screenshot added to this PR, but on my device and emulator, the fix are OK.

PR also contains:
- clear cache reset some stores, to be able to see the Welcome Screen again;
- some small cleanup to the related classes.

Fixes #892.

### CODE REVIEW

File: features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt [hunk 1]
```diff
@@ -43,6 +44,7 @@ class DefaultClearCacheUseCase @Inject constructor(
     private val coroutineDispatchers: CoroutineDispatchers,
     private val defaultCacheIndexProvider: DefaultCacheService,
     private val okHttpClient: Provider<OkHttpClient>,
+    private val ftueState: Provider<FtueState>,
```
Comment (bmarty) [inline features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt:47]: It's compiling, but crashing in the room list if I use the `FtueState` type directly, not sure why :/

File: features/ftue/impl/src/main/kotlin/io/element/android/features/ftue/impl/welcome/WelcomeView.kt [hunk 2]
```diff
@@ -75,13 +75,15 @@ fun WelcomeView(
                 Text(
                     modifier = Modifier.testTag(TestTags.welcomeScreenTitle),
                     text = stringResource(R.string.screen_welcome_title, applicationName),
-                    style = ElementTheme.typography.fontHeadingLgBold,
+                    style = ElementTheme.typography.fontHeadingMdBold,
```
Comment (bmarty) [inline features/ftue/impl/src/main/kotlin/io/element/android/features/ftue/impl/welcome/WelcomeView.kt:78]: Changing the font, see in [Figma](https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?node-id=8029%3A128756&mode=dev). (the iOS Figma was used I guess)

File: features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt (hunk 1 above)
Comment (jmartinesp) [reply features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt:47]: It's working fine for me without the `Provider`. Maybe I'm doing something wrong? What did you do reproduce the issue?

**Review by jmartinesp — APPROVED** [review]
LGTM, just a question. Also, I wonder if resetting the FTUE state when clearing cache is the best behavior, but it's certainly useful for testing.

File: features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt (hunk 1 above)
Comment (bmarty) [reply features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt:47]: I will try again, but I had the classical `Cannot find NodeFactory for RoomFlowNode` issue.

Comment (bmarty) [reply features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/tasks/ClearCacheUseCase.kt]: It's working today, I have dropped the commit which introduced this.

**Pull request merged**